### PR TITLE
Better PROXY_PROTOCOL

### LINKED ISSUE #3162: snappymail integration broken with TLS_FLAVOR=notls in master
- Version: `master`

## Description
Using snappymail as webmail is broken in latest master. Trying to login to the webmail results in a redirect loop.
Even if logged in sending new mails fails. Also managing sieve rules is not possible due to a server error.

## Replication Steps
1. Create a clean new mailu instance using stable 2.0 images using snappymail as webmail (image `ghcr.io/mailu/webmail:2.0`)
2. Login to webmail and verify everthing works as expected
3. Change webmail image from 2.0 tag to master (i.e. use image `ghcr.io/mailu/webmail:master`)
4. Try to login to webmail again and observe described errors

## Possible Fix
I narrowed the origin of the regression down to f143aa3dc8d8ac6d9b8ca34d13b22d90318c4523: changing `secure` back to `0` in `default.json` fixes the described issues:
~~~diff
diff --git a/webmails/snappymail/defaults/default.json b/webmails/snappymail/defaults/default.json
index 12252aee..3d8efaa2 100644
--- a/webmails/snappymail/defaults/default.json
+++ b/webmails/snappymail/defaults/default.json
@@ -3,7 +3,7 @@
     "IMAP": {
         "host": "{{ FRONT_ADDRESS }}",
         "port": 10143,
-        "secure": 2,
+        "secure": 0,
         "shortLogin": false,
         "ssl": {
             "verify_peer": false,
@@ -20,7 +20,7 @@
     "SMTP": {
         "host": "{{ FRONT_ADDRESS }}",
         "port": 10025,
-        "secure": 2,
+        "secure": 0,
         "shortLogin": false,
         "ssl": {
             "verify_peer": false,
@@ -37,7 +37,7 @@
     "Sieve": {
         "host": "{{ FRONT_ADDRESS }}",
         "port": 14190,
-        "type": 2,
+        "secure": 0,
         "shortLogin": false,
         "ssl": {
             "verify_peer": false,
~~~

But I am not sure about the implications of this change.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

Feature

## What does this PR do?

- Disable IMAP, POP3 and Submission by default; see https://nostarttls.secvuln.info/ on why explicit TLS is going away.
- Change the semantic of PROXY_PROTOCOL to make it configurable per port
- fix TLS_FLAVOR=notls not working with snappymail
- fix TLS_PERMISSIVE
- remove KUBERNETES_INGRESS; shouldn't be needed anymore
- update the documentation and the reverse proxy example

### Related issue(s)
- close #3162
- close #3061

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: core/nginx/conf/nginx.conf [hunk 1]
```diff
@@ -95,18 +95,18 @@ http {
       client_max_body_size {{ MESSAGE_SIZE_LIMIT|int + 8388608 }};
 
       # Listen on HTTP only in kubernetes or behind reverse proxy
-      {% if KUBERNETES_INGRESS or TLS_FLAVOR in [ 'mail-letsencrypt', 'notls', 'mail' ] %}
-      listen 80{% if PROXY_PROTOCOL in ['all', 'http'] %} proxy_protocol{% endif %};
+      {% if TLS_FLAVOR in [ 'mail-letsencrypt', 'notls', 'mail' ] %}
+      listen 80{% if PROXY_HTTPPROTOCOL_80 %} proxy_protocol{% endif %};
```
Comment (Diman0) [inline core/nginx/conf/nginx.conf:99]: This is probably a typo and should be `PROXY_PROTOCOL_80`

```suggestion
      listen 80{% if PROXY_PROTOCOL_80 %} proxy_protocol{% endif %};
```

File: core/nginx/dovecot/proxy.conf [hunk 2]
```diff
@@ -142,20 +148,25 @@ service lmtp {
...
 {% endif %}
    }
+{%- if TLS_465 and PORT_465 %}
    inet_listener submissions {
     port = 465
-{%- if TLS %}
     ssl = yes
-{% endif %}
-{%- if PROXY_PROTOCOL in ['all', 'all-but-http', 'mail'] %}
+{%- if PROXY_PROTOCOL_645 %}
```
Comment (Diman0) [inline core/nginx/dovecot/proxy.conf:165]: Typo

```suggestion
{%- if PROXY_PROTOCOL_465 %}
```

**Review by Diman0 — CHANGES_REQUESTED** [review]
It looks good to me. My current setup with proxy_protocol still works fine with this PR.
There are only two typo's that must be addressed. See in-line comments.

**Comment by nextgens** [comment]
Great catch; not sure how I have managed to typo that.

**Review by Diman0 — APPROVED** [review]
LGTM

**Pull request merged**